tests/unit_test/elife_article_xml/elife_article_xml_pipeline_test.py

```diff
@@ -0,0 +1,339 @@
+        actual_return_value = list(iter_unprocessed_xml_file_url_from_git_directory(
+            source_config=SOURCE_CONFIG_1,
+            processed_file_url_list=PROCESSED_FILE_URL_LIST
+        ))
+        assert actual_return_value == [NOT_EMPTY_ARTICLE_URL_1, NOT_EMPTY_ARTICLE_URL_2]
+
+    def test_should_return_all_url_list_if_processed_files_list_empty(
+        self,
+        get_json_response_from_url_mock: MagicMock
+    ):
+        get_json_response_from_url_mock.return_value = GITHUB_TREE_REPONSE_2
+        actual_return_value = list(iter_unprocessed_xml_file_url_from_git_directory(
+            source_config=SOURCE_CONFIG_1,
+            processed_file_url_list=[]
+        ))
+        assert actual_return_value == [
+            NOT_EMPTY_ARTICLE_URL_1,
+            NOT_EMPTY_ARTICLE_URL_2,
+            PROCESSED_NOT_EMPTY_ARTICLE_URL
+        ]
+
+
+class TestIterDecodedXmlFileContent:
+    def test_should_return_decoded_file_content(
+        self,
+        get_json_response_from_url_mock: MagicMock
+    ):
+        get_json_response_from_url_mock.return_value = XML_FILE_JSON
+        actual_return_value = list(iter_decoded_xml_file_content(XML_FILE_URL_LIST))
+        assert actual_return_value == [XML_FILE_CONTENT_DECODED]
+
+
+class TestIterBqCompatibleJson:
+    def test_should_keep_compatible_key_name_with_underscore(self):
+        input_data = [{
+            'key_1': 'value1'
+        }]
+        result = get_bq_compatible_json_dict(input_data)
+        assert result == input_data
+
+    def test_should_replace_hyphen_in_key_name_with_underscore(self):
+        input_data = [{
+            'key-1': 'value1'
+        }]
+        expected_data = [{
+            'key_1': 'value1'
+        }]
+        result = get_bq_compatible_json_dict(input_data)
+        assert result == expected_data
+
+    def test_should_remove_none_value(self):
+        input_data = [{
+            'key_1': None
+        }]
+        expected_data = []
+        result = get_bq_compatible_json_dict(input_data)
+        assert result == expected_data
+
+    def test_should_remove_empty_list_value(self):
+        input_data = [{
+            'key_1': []
+        }]
+        expected_data = []
+        result = get_bq_compatible_json_dict(input_data)
+        assert result == expected_data
+
+
+class TestGetArticleJsonDataFromXmlStringContent:
+    def test_should_return_empty_dict_if_the_xml_with_empty_root(self):
+        xml_string = '<article></article>'
+        return_value = get_article_json_data_from_xml_string_content(xml_string)
+        assert return_value == {}
+
+    def test_should_return_empty_dict_if_the_xml_has_empty_article_meta(self):
+        xml_string = '<article><front><article-meta></article-meta></front></article>'
+        return_value = get_article_json_data_from_xml_string_content(xml_string)
+        assert return_value == {}
+
+    def test_should_return_dict_with_related_article_if_it_exists(self):
```

Comment:
We should add a test with multiple related articles as I suspect there will be an issue with the href namespace.